File: mvn-advent-of-code/src/test/java/advent/twenty_eighteen/Day07TheSumOfItsPartsTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day07TheSumOfItsPartsTest {
    @Test
    public void checkReal() {
        Day07TheSumOfItsParts im = new Day07TheSumOfItsParts("/twenty_eighteen/Day07-TheSumOfItsParts-input.txt", 5, 60);

        assertEquals("ABLCFNSXZPRHVEGUYKDIMQTWJO", im.getPart1Answer());
        assertEquals(1157, im.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        Day07TheSumOfItsParts im = new Day07TheSumOfItsParts("/twenty_eighteen/Day07-TheSumOfItsParts-example#1.txt", 2, 0);

        assertEquals("CABDFE", im.getPart1Answer());
        assertEquals(15, im.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day07TheSumOfItsPartsTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day07TheSumOfItsPartsTest  extends FlatSpec with Matchers {
  "2018 Day 07 - Input" should "calculate answers" in {
    val dp = new Day07TheSumOfItsParts("/twenty_eighteen/Day07-TheSumOfItsParts-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day07TheSumOfItsParts("/twenty_eighteen/Day07-TheSumOfItsParts-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
